### DISCUSSION TITLE
Common output format

### DISCUSSION THREAD

**Discussion by telyn** [robbrad/UKBinCollectionData#143] [Ideas]
Heya! Thanks for merging my PR earlier :) 

I wanted to float the idea of a common output format - I was thinking this would be an optional addition that would have to be enabled by a flag on the command line.

This would make it easy to swap council when one moves house without having to rework the downstream consumer. 

My first thought is implementing it as an extra method on each council class which takes an object matching that council’s schema (checking that it does!) and returns one in the common format.

That way we can unit test that bit, asserting on the way in and out that the object matches the relevant schema. 

How does that sound?

**Comment by dp247** [comment]
I'd certainly be interested to see a proof of concept 😁

**Reply by telyn** [reply]
😉 I’ll try to make my function and variable names sensible then.

**Comment by OliverCullimore** [comment]
@dp247 @telyn Just discovered this discussion, I've kinda gone and moved all councils to use a common output format as part of #296 already as I realised when we come to look at #228 it would make things a lot easier.

I noted it would be a breaking change if someone re cloned the project and was consuming one of the alternative formats before but if we enforce this common output format is always used going forwards it should be a one-off breaking change that improves things overall.
